hw/ip_templates/rv_core_ibex/doc/boot-rom-patching.md

```diff
@@ -0,0 +1,476 @@
+# Boot, ROM execution and Patching
+
+## Glossary
+
+| Term  | Description                         |
+|-------|-------------------------------------|
+| SoC   | System On Chip                      |
+| DFx   | Design for Test and Debug           |
+| DFT   | Design for Test                     |
+
+## Scope
+
+This document covers the boot mechanism for Integrated OpenTitan.
+This includes the execution of the OpenTitan ROM, associated patch support in efuses / OTP for survivability and the code fetch from external flash.
+
+## Overview
+
+The integrated Open Titan is a root of trust for measurement, integrated within the SoC.
+It provides the framework necessary to bootstrap the SoC computing elements in a secure and trusted fashion.
+This document covers the changes necessary for the OpenTitan to boot first while also allowing an SoC with integrated OpenTitan to add small customizations to the boot path configuration that fit with the SoC topology.
+A few examples are voltage ramp up sequence, clock settings etc.
+Such customizations are expected to
+
+- Be non-proprietary such that they can be made available in the open source environment for audit.
+- Only require minimal flow to enable a path to external flash that is under OpenTitan control.
+
+Flows involving reset and bring up for the rest of the SoC, which may be potentially more complex and involve proprietary code, are expected to be handled by other computing element(s) that are bootstrapped by the integrated OpenTitan and running authenticated / measured code.
+
+This proposal introduces a second OpenTitan boot ROM partition with efuse/OTP based patching mechanism that enables each SoC to add such small customizations and provide critical survivability hooks while also removing the dependency on the integrated flash component.
```

Comment:
Another nit: "... while also removing the dependency ..." sounds like we're talking about a particular dependency. I don't think I know an example at the moment? I think we can tweak the text and avoid a long sentence with something like this?
```
This proposal introduces a second OpenTitan boot ROM partition with an efuse/OTP based patching mechanism.
This combination enables each SoC to add such small customizations and provide critical survivability hooks without needing an integrated flash component.
```